pkg/virt-controller/watch/clone/clone.go

```diff
@@ -178,7 +179,7 @@ func (ctrl *VMCloneController) syncSourceVMTargetVM(key string, source *k6tv1.Vi
 		snapshot = obj.(*snapshotv1alpha1.VirtualMachineSnapshot)
 		logger.Infof("found snapshot %s for clone %s", snapshot.Name, vmClone.Name)
 
-		if snapshot.Status.ReadyToUse == nil || *snapshot.Status.ReadyToUse == false {
+		if snapshot.Status == nil || snapshot.Status.ReadyToUse == nil || *snapshot.Status.ReadyToUse == false {
```

Comment:
maybe use existing function of checking if snapshot is ready? vmSnapshotReady
and also for restore you have function -> vmRestoreProgressing to use in line 239

Reply:
Done, thanks!